metadoc/js/src/main/resources/scala.ts

```diff
@@ -0,0 +1,157 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+// Based on https://github.com/Microsoft/monaco-languages/blob/3e7008e334179c192b1cc3684475a110139d4662/src/java.ts
+import * as monaco from 'vs/editor/editor.main';
+
+import IRichLanguageConfiguration = monaco.languages.LanguageConfiguration;
+import ILanguage = monaco.languages.IMonarchLanguage;
+
+export const conf : IRichLanguageConfiguration = {
+	// the default separators except `@$`
+	wordPattern: /(-?\d*\.\d\w*)|([^\`\~\!\#\%\^\&\*\(\)\-\=\+\[\{\]\}\\\|\;\:\'\"\,\.\<\>\/\?\s]+)/g,
+	comments: {
+		lineComment: '//',
+		blockComment: ['/*', '*/'],
+	},
+	brackets: [
+		['{', '}'],
+		['[', ']'],
+		['(', ')'],
+	],
+	autoClosingPairs: [
+		{ open: '{', close: '}' },
+		{ open: '[', close: ']' },
+		{ open: '(', close: ')' },
+		{ open: '"', close: '"' },
+		{ open: '\'', close: '\'' },
+	],
+	surroundingPairs: [
+		{ open: '{', close: '}' },
+		{ open: '[', close: ']' },
+		{ open: '(', close: ')' },
+		{ open: '"', close: '"' },
+		{ open: '\'', close: '\'' },
+		{ open: '<', close: '>' },
+	]
+};
+
+export const language = <ILanguage>{
+	defaultToken: '',
+	tokenPostfix: '.scala',
+
+	keywords: [
+		// From https://www.scala-lang.org/files/archive/spec/2.11/01-lexical-syntax.html
+		'abstract', 'case', 'catch', 'class', 'def',
+		'do', 'else', 'extends', 'false', 'final',
+		'finally', 'for', 'forSome', 'if', 'implicit',
+		'import', 'lazy', 'macro', 'match', 'new',
+		'null', 'object', 'override', 'package', 'private',
+		'protected', 'return', 'sealed', 'super', 'this',
+		'throw', 'trait', 'try', 'true', 'type',
+		'val', 'var', 'while', 'with', 'yield',
+	],
+
+	operators: [
+		'_', ':', '=', '=>', '<-', '<:', '<%', '>:', '#', '@',
+
+		// Copied from java.ts, to be validated
+		'=', '>', '<', '!', '~', '?', ':',
```

Comment:
These might be some help https://github.com/scalameta/scalameta/blob/8a4192713f98d177773f8c835c771ddc71d9901c/scalameta/tokens/shared/src/main/scala/scala/meta/tokens/Token.scala#L70